var/spack/repos/builtin/packages/nalu-wind/package.py

```diff
@@ -85,6 +86,11 @@ class NaluWind(CMakePackage, CudaPackage):
 
     depends_on('boost cxxstd=14', when='+boost')
 
+    # TODO: replace this with an explicit list of components of Boost,
+    # for instance depends_on('boost +filesystem')
+    # See https://github.com/spack/spack/pull/22303 for reference
+    depends_on(Boost.with_default_variants, when='+boost')
```

Comment:
Given @sayerhs's feedback, I guess the update here is:
```suggestion
    depends_on('boost +filesystem +iostreams cxxstd=14', when='+boost')
```